python/CLOUDTRAIL_S3_DATAEVENTS_ENABLED/CLOUDTRAIL_S3_DATAEVENTS_ENABLED.py

```diff
@@ -0,0 +1,488 @@
+     Then: Return COMPLIANT
+'''
+import json
+import sys
+import datetime
+import re
+import boto3
+import botocore
+
+try:
+    import liblogging
+except ImportError:
+    pass
+
+##############
+# Parameters #
+##############
+
+# Define the default resource to report to Config Rules
+DEFAULT_RESOURCE_TYPE = 'AWS::::Account'
+
+# Set to True to get the lambda to assume the Role attached on the Config Service (useful for cross-account).
+ASSUME_ROLE_MODE = False
+
+# Other parameters (no change needed)
+CONFIG_ROLE_TIMEOUT_SECONDS = 900
+DEFAULT_CHECK = 'arn:aws:s3'
+
+
+#############
+# Main Code #
+#############
+
+def evaluate_compliance(event, configuration_item, valid_rule_parameters):
+    ct_client = get_client('cloudtrail', event)
+    evaluations = []
+    compliant_buckets = []
+    non_compliant_buckets = []
+    trail_list = get_all_trails(ct_client)
+    region = event["configRuleArn"][15:].split(':')[0]
+
+    #No trails configured: return NON_COMPLIANT
+    if not trail_list:
+        evaluations.append(build_evaluation(event['accountId'], 'NON_COMPLIANT', event, annotation='No AWS CloudTrail Trails are present to log events in this region.'))
+        return evaluations
+
+    #No rule parameter specified, default check for all S3 buckets
+    if not valid_rule_parameters:
+        for trail in trail_list:
+            trail_region = trail['HomeRegion']
+            if trail_region == region:
+                if trail['HasCustomEventSelectors']:
+                    event_selector_output = ct_client.get_event_selectors(TrailName=trail['Name'])['EventSelectors']
+                    data_resource_values = get_data_resource_s3_values(event_selector_output)
+                    if isinstance(data_resource_values, bool):
+                        evaluations.append(build_evaluation(event['accountId'], 'COMPLIANT', event))
+                        return evaluations
+
+    #Rule parameter S3BucketName is specified
+    else:
+        for trail in trail_list:
+            trail_region = trail['HomeRegion']
+            if trail_region == region:
+                if trail['HasCustomEventSelectors']:
+                    event_selector_output = ct_client.get_event_selectors(TrailName=trail['Name'])['EventSelectors']
+                    data_resource_values = get_data_resource_s3_values(event_selector_output)
+                    #if logging is enabled for 'arn:aws:s3', return COMPLIANT
+                    if isinstance(data_resource_values, bool):
+                        evaluations.append(build_evaluation(event['accountId'], 'COMPLIANT', event))
+                        return evaluations
+
+                #get bucket names from data resource values to store them in a list
+                for value in data_resource_values:
+                    # value is of the form arn:aws:s3:::test/, [13:-1] strips 'arn:aws:s3:::' and the trailing slash
+                    bucket_name = value[13:-1]

```

Comment:
This is going to be challenging for other partitions (china, gov-cloud, etc.).

Reply:
Agreed, will try to split using ':' and extract value using the split.

Reply:
Need update?